src/constructs/ecsIsoServiceAutoscaler/ecsIsoServiceAutoscaler.ts

```diff
@@ -9,16 +9,46 @@ import { Alarm } from 'aws-cdk-lib/aws-cloudwatch';
 import { Cluster, IService } from 'aws-cdk-lib/aws-ecs';
 import { Rule, Schedule } from 'aws-cdk-lib/aws-events';
 import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
-import { IRole } from 'aws-cdk-lib/aws-iam';
+import { Effect, IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
 import { Code, Function, Runtime } from 'aws-cdk-lib/aws-lambda';
 import { Construct } from 'constructs';
 
 export interface EcsIsoServiceAutoscalerProps {
   /**
-   * The IAM role that allows the created lambda to adjust the desired count on the ECS Service .
-   * TODO: Make optional and add function for auto generation
+   * Optional IAM role to attach to the created lambda to adjust the desired count on the ECS Service
+   *
+   * Ensure this role has appropriate privileges. Example IAM policy statements:
+   * And more
+   * ```json
+   * {
+   *  "PolicyDocument": {
+   *    "Statement": [
+   *      {
+   *        "Action": "cloudwatch:DescribeAlarms",
+   *        "Effect": "Allow",
+   *        "Resource": "arn:${Partition}:cloudwatch:${Region}:${Account}:alarm:${AlarmName}"
+   *      },
+   *      {
+   *        "Action": [
+   *          "ecs:DescribeServices",
+   *          "ecs:UpdateService"
+   *        ],
+   *        "Condition": {
+   *          "StringEquals": {
+   *            "ecs:cluster": "arn:${Partition}:ecs:${Region}:${Account}:cluster/${ClusterName}"
+   *          }
+   *        },
+   *        "Effect": "Allow",
+   *        "Resource": "arn:${Partition}:ecs:${Region}:${Account}:service/${ClusterName}/${ServiceName}"
+   *      }
+   *    ],
+   *    "Version": "2012-10-17"
+   *  }
+   *}
+   * ```
```

Comment:
Would a CDK-native version of a [Role](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.Role.html) with the right [PolicyDocument](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.PolicyDocument.html) more sense?

Reply:
The original implementation had a role passed into it to attach to the Lambda function that handled the autoscaling.  The idea being that the enterprise didn't allow developers to create IAM roles, so they'd have to pass a provided role in.

This explanation of what IAM policy is needed for the Lambda function to do what it needs is more for a developer to provide to a centralized team in the enterprise who creates these roles for them